### ISSUE TITLE
Controller from require is empty when using an explicit return

### ISSUE THREAD

**Issue by MerlierR** [angular/angular.js#11147]
<strong>AngularJS v1.3.13</strong>

When a directive uses ``require: '^outer'``, the injected controller is ``{}`` in case an explicit return is used. (``controller: function () { return ...; }``)

<strong>This issue does not occur in angular version 1.2.28</strong>

Example: http://jsfiddle.net/s6oyL2xk/
```
var app = angular.module('myApp',[]);

app.directive('outer', function () {
    return {
        restrict: 'E',
        template: '<div>Outer directive</div><div ng-transclude></div>',
        transclude: true,
        controller: function OuterController() {   
            return {
                data: 'outerController data'
            }
        }
    }
});

app.directive('inner', function () {
    return {
        restrict: 'E',
        template: '<div>Inner Directive</div><div ng-bind="test"></div>',
        require: '^outer',
        link: function (scope, element, attrs, outerController) {
            _.isEmpty(outerController) === true; // This should not happen
            scope.test = outerController.data;
        }
    }
});
```

**Comment by gkalpak** [comment]
This is due to a breaking changed introduced in v1.2.17/v1.3.0-beta.10 (see relevant [changelog entry][1]).

> $compile: due to 5f3f25a1,
The returned value from directive controller constructors are now ignored, and only the constructed instance itself will be attached to the node's expando.

--
Since v1.4.0-beta.3 the previous behaviour is restored and the return value will be used instead (see relevant [changelog entry][2]).
(But I doubt it is going to be backported to v1.3.x.)

[1]: https://github.com/angular/angular.js/blob/master/CHANGELOG.md#since-1217-and-130-beta10
[2]: https://github.com/angular/angular.js/blob/master/CHANGELOG.md#140-beta3-substance-mimicry-2015-02-02

**Comment by gkalpak** [comment]
Hm...I just noticed that whether theoretically my comment above should be valid, in the case of directive controllers, things work differently. (The behaviour is rather unexpected/buggy.)

**Comment by pkozlowski-opensource** [comment]
Indeed, this doesn't sound right / expected, putting it in the backlog. At the same time people returning things from a constructor function are asking for trouble :-)

**Comment by jamestalmage** [comment]
Fix in PR #11326
